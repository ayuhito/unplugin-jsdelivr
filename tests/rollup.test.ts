import * as path from "pathe"
import { rollup } from "rollup"
import { describe, expect, it } from "vitest"

import jsDelivr from '../src/rollup'

describe('Rollup build', () => {

  it('should rewrite imports for lodash', async () => {
    const bundle = await rollup({
      input: "./tests/fixtures/lodash.ts",
      plugins: [jsDelivr({ cwd: path.join(process.cwd(), 'tests/fixtures'), modules: [{ module: 'lodash' }] })]
    })
    const { output } = await bundle.generate({ format: 'esm' })
    expect(output[0].imports).toEqual(['https://cdn.jsdelivr.net/npm/lodash@4.17.21/+esm'])
  })

  it('should skip rewrite imports for lodash', async () => {
    const bundle = await rollup({
      input: './tests/fixtures/lodash.ts',
      plugins: [jsDelivr({ cwd: path.join(process.cwd(), 'tests/fixtures'), modules: [] })]
    })
    const { output } = await bundle.generate({ format: 'esm' })
    expect(output[0].imports).toEqual(['lodash'])
  })
})
