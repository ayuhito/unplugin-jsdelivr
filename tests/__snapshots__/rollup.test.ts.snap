// Vitest Snapshot v1

exports[`Rollup build > No transform > should rewrite imports for basic 1`] = `
[
  {
    "code": "import map from 'https://cdn.jsdelivr.net/npm/lodash@4.17.21/+esm';
import colors from 'picocolors';
import Undermap from 'https://cdn.jsdelivr.net/npm/underscore@1.13.4/+esm';

const testMap = map([1, 2, 3], x => x + 1);
const testMerge = map({ a: 1 }, { b: 2 });
const testColors = colors.bold('test');
const testMap2 = Undermap([1, 2, 3], x => x + 1);

export { testColors, testMap, testMap2, testMerge };
",
    "dynamicImports": [],
    "exports": [
      "testColors",
      "testMap",
      "testMap2",
      "testMerge",
    ],
    "facadeModuleId": "C:\\\\Users\\\\spydo\\\\Desktop\\\\Websites\\\\unplugin-jsdelivr\\\\tests\\\\fixtures\\\\basic.ts",
    "fileName": "basic.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/+esm": [
        "default",
      ],
      "https://cdn.jsdelivr.net/npm/underscore@1.13.4/+esm": [
        "default",
      ],
      "picocolors": [
        "default",
      ],
    },
    "imports": [
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/+esm",
      "picocolors",
      "https://cdn.jsdelivr.net/npm/underscore@1.13.4/+esm",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "C:\\\\Users\\\\spydo\\\\Desktop\\\\Websites\\\\unplugin-jsdelivr\\\\tests\\\\fixtures\\\\basic.ts": {
        "code": "const testMap = map([1, 2, 3], x => x + 1);
const testMerge = map({ a: 1 }, { b: 2 });
const testColors = colors.bold('test');
const testMap2 = Undermap([1, 2, 3], x => x + 1);",
        "originalLength": 370,
        "removedExports": [],
        "renderedExports": [
          "testColors",
          "testMap",
          "testMap2",
          "testMerge",
        ],
        "renderedLength": 176,
      },
    },
    "name": "basic",
    "referencedFiles": [],
    "type": "chunk",
  },
]
`;

exports[`Rollup build > No transform > should skip rewrite imports for basic 1`] = `
[
  {
    "code": "import { map, merge } from 'lodash';
import colors from 'picocolors';
import { map as map$1 } from 'underscore';

const testMap = map([1, 2, 3], x => x + 1);
const testMerge = merge({ a: 1 }, { b: 2 });
const testColors = colors.bold('test');
const testMap2 = map$1([1, 2, 3], x => x + 1);

export { testColors, testMap, testMap2, testMerge };
",
    "dynamicImports": [],
    "exports": [
      "testColors",
      "testMap",
      "testMap2",
      "testMerge",
    ],
    "facadeModuleId": "C:\\\\Users\\\\spydo\\\\Desktop\\\\Websites\\\\unplugin-jsdelivr\\\\tests\\\\fixtures\\\\basic.ts",
    "fileName": "basic.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "lodash": [
        "map",
        "merge",
      ],
      "picocolors": [
        "default",
      ],
      "underscore": [
        "map",
      ],
    },
    "imports": [
      "lodash",
      "picocolors",
      "underscore",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "C:\\\\Users\\\\spydo\\\\Desktop\\\\Websites\\\\unplugin-jsdelivr\\\\tests\\\\fixtures\\\\basic.ts": {
        "code": "const testMap = map([1, 2, 3], x => x + 1);
const testMerge = merge({ a: 1 }, { b: 2 });
const testColors = colors.bold('test');
const testMap2 = map$1([1, 2, 3], x => x + 1);",
        "originalLength": 370,
        "removedExports": [],
        "renderedExports": [
          "testColors",
          "testMap",
          "testMap2",
          "testMerge",
        ],
        "renderedLength": 175,
      },
    },
    "name": "basic",
    "referencedFiles": [],
    "type": "chunk",
  },
]
`;

exports[`Rollup build > With transform > should split imports and rewrite for basic 1`] = `
[
  {
    "code": "import map from 'https://cdn.jsdelivr.net/npm/lodash@4.17.21/map/+esm';
import LodashMerge from 'https://cdn.jsdelivr.net/npm/lodash@4.17.21/merge/+esm';
import colors from 'picocolors';
import { map as map$1 } from 'underscore';

const testMap = map([1, 2, 3], x => x + 1);
const testMerge = LodashMerge({ a: 1 }, { b: 2 });
const testColors = colors.bold('test');
const testMap2 = map$1([1, 2, 3], x => x + 1);

export { testColors, testMap, testMap2, testMerge };
",
    "dynamicImports": [],
    "exports": [
      "testColors",
      "testMap",
      "testMap2",
      "testMerge",
    ],
    "facadeModuleId": "C:\\\\Users\\\\spydo\\\\Desktop\\\\Websites\\\\unplugin-jsdelivr\\\\tests\\\\fixtures\\\\basic.ts",
    "fileName": "basic.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/map/+esm": [
        "default",
      ],
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/merge/+esm": [
        "default",
      ],
      "picocolors": [
        "default",
      ],
      "underscore": [
        "map",
      ],
    },
    "imports": [
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/map/+esm",
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/merge/+esm",
      "picocolors",
      "underscore",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "C:\\\\Users\\\\spydo\\\\Desktop\\\\Websites\\\\unplugin-jsdelivr\\\\tests\\\\fixtures\\\\basic.ts": {
        "code": "const testMap = map([1, 2, 3], x => x + 1);
const testMerge = LodashMerge({ a: 1 }, { b: 2 });
const testColors = colors.bold('test');
const testMap2 = map$1([1, 2, 3], x => x + 1);",
        "originalLength": 370,
        "removedExports": [],
        "renderedExports": [
          "testColors",
          "testMap",
          "testMap2",
          "testMerge",
        ],
        "renderedLength": 181,
      },
    },
    "name": "basic",
    "referencedFiles": [],
    "type": "chunk",
  },
]
`;

exports[`Rollup build > With transform > should split multiple imports and rewrite for basic 1`] = `
[
  {
    "code": "import map from 'https://cdn.jsdelivr.net/npm/lodash@4.17.21/map/+esm';
import LodashMerge from 'https://cdn.jsdelivr.net/npm/lodash@4.17.21/merge/+esm';
import colors from 'picocolors';
import Undermap from 'https://cdn.jsdelivr.net/npm/underscore@1.13.4/lib/map/+esm';

const testMap = map([1, 2, 3], x => x + 1);
const testMerge = LodashMerge({ a: 1 }, { b: 2 });
const testColors = colors.bold('test');
const testMap2 = Undermap([1, 2, 3], x => x + 1);

export { testColors, testMap, testMap2, testMerge };
",
    "dynamicImports": [],
    "exports": [
      "testColors",
      "testMap",
      "testMap2",
      "testMerge",
    ],
    "facadeModuleId": "C:\\\\Users\\\\spydo\\\\Desktop\\\\Websites\\\\unplugin-jsdelivr\\\\tests\\\\fixtures\\\\basic.ts",
    "fileName": "basic.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/map/+esm": [
        "default",
      ],
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/merge/+esm": [
        "default",
      ],
      "https://cdn.jsdelivr.net/npm/underscore@1.13.4/lib/map/+esm": [
        "default",
      ],
      "picocolors": [
        "default",
      ],
    },
    "imports": [
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/map/+esm",
      "https://cdn.jsdelivr.net/npm/lodash@4.17.21/merge/+esm",
      "picocolors",
      "https://cdn.jsdelivr.net/npm/underscore@1.13.4/lib/map/+esm",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "C:\\\\Users\\\\spydo\\\\Desktop\\\\Websites\\\\unplugin-jsdelivr\\\\tests\\\\fixtures\\\\basic.ts": {
        "code": "const testMap = map([1, 2, 3], x => x + 1);
const testMerge = LodashMerge({ a: 1 }, { b: 2 });
const testColors = colors.bold('test');
const testMap2 = Undermap([1, 2, 3], x => x + 1);",
        "originalLength": 370,
        "removedExports": [],
        "renderedExports": [
          "testColors",
          "testMap",
          "testMap2",
          "testMerge",
        ],
        "renderedLength": 184,
      },
    },
    "name": "basic",
    "referencedFiles": [],
    "type": "chunk",
  },
]
`;
